common/src/View/MapViewBase.cpp

```diff
@@ -1031,6 +1032,8 @@ void MapViewBase::doRender()
   renderContext.setShowFog(pref(Preferences::ShowFog));
   renderContext.setShowGrid(grid.visible());
   renderContext.setGridSize(grid.actualSize());
+  const auto ratio = QApplication::primaryScreen()->devicePixelRatio(); // to be changed
```

Comment:
Still using `primaryScreen()` for now, as I have no idea how to get the main QWindow. @kduske ideas?
I have tried:
```
const auto window1 = QApplication::focusWindow(); // NULL when using file picker
const auto window2 = QApplication::activeWindow(); // NULL when using file picker
const auto window3 = QWindow::fromWinId(0); // works OK, but am I doing the right thing?
```

Reply:
Don't use `QScreen` at all. instead call `window()->devicePixelRatio();`

Reply:
Oh, and I think you can just inline the `ratio` variable.